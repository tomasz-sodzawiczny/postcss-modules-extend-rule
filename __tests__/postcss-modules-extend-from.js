import extendFrom from "../src/plugins/postcss-modules-extend-from";

const process = css => extendFrom.process(css).css;

describe("postcss-modules-extend-from", () => {
  it("works with a single extend", () => {
    const css = ".a { @extend .red from 'colors'; }";

    expect(process(css)).toMatchSnapshot();
  });

  it("works with multiple classes extending single class", () => {
    const css =
      ".a { @extend .red from 'colors'; } .b { @extend .red from 'colors'; }";

    expect(process(css)).toMatchSnapshot();
  });

  it("works with multiple classes extending multiple classes from one file", () => {
    const css =
      ".a { @extend .red from 'colors'; } .b { @extend .blue from 'colors'; }";

    expect(process(css)).toMatchSnapshot();
  });

  it("works with extending classes from multiple files", () => {
    const css = ".a { @extend .red from 'red'; @extend .blue from 'blue';}";

    expect(process(css)).toMatchSnapshot();
  });
});
