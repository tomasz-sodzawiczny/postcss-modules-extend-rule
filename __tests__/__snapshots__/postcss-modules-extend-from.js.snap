// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-modules-extend-from works with a single extend 1`] = `
":import('colors') {
  __extends_from_1__red: red
}
.a { @postcss-modules-extend-class __extends_from_1__red; }"
`;

exports[`postcss-modules-extend-from works with extending classes from multiple files 1`] = `
":import('red') {
  __extends_from_5__red: red
}
:import('blue') {
  __extends_from_6__blue: blue
}
.a { @postcss-modules-extend-class __extends_from_5__red; @postcss-modules-extend-class __extends_from_6__blue;}"
`;

exports[`postcss-modules-extend-from works with multiple classes extending multiple classes from one file 1`] = `
":import('colors') {
  __extends_from_3__red: red;
  __extends_from_4__blue: blue
} .a { @postcss-modules-extend-class __extends_from_3__red; } .b { @postcss-modules-extend-class __extends_from_4__blue; }"
`;

exports[`postcss-modules-extend-from works with multiple classes extending single class 1`] = `
":import('colors') {
  __extends_from_2__red: red
} .a { @postcss-modules-extend-class __extends_from_2__red; } .b { @postcss-modules-extend-class __extends_from_2__red; }"
`;
